client/app/api/course/Materials.ts

```diff
@@ -15,8 +17,12 @@ const getShouldDownloadFromContentDisposition = (
 };
 
 export default class MaterialsAPI extends BaseCourseAPI {
+  get #materialPrefix(): string {
```

Comment:
Similarly, you shouldn't be introducing folder logic into `MaterialsAPI`, it should go into `client/app/api/course/MaterialFolders.js` instead.  

At the same time, good to rewrite the file into TypeScript.